drivers/digitalocean/digitalocean.go

```diff
@@ -377,19 +377,21 @@ func (d *Driver) Kill() error {
 
 func (d *Driver) Remove() error {
 	client := d.getClient()
-	if d.SSHKeyFingerprint == "" {
+	if d.SSHKeyFingerprint == "" && d.SSHKeyID != 0 {
 		if resp, err := client.Keys.DeleteByID(context.TODO(), d.SSHKeyID); err != nil {
 			if resp.StatusCode == 404 {
 				log.Infof("Digital Ocean SSH key doesn't exist, assuming it is already deleted")
 			} else {
+				log.Errorf("ERROR: %v", err)
 				return err
 			}
 		}
 	}
 	if resp, err := client.Droplets.Delete(context.TODO(), d.DropletID); err != nil {
-		if resp.StatusCode == 404 {
+		if resp != nil && resp.StatusCode == 404 {
 			log.Infof("Digital Ocean droplet doesn't exist, assuming it is already deleted")
 		} else {
+			log.Errorf("ERROR: %v", err)
```

Comment:
either drop this and let the error bubble up or make it more descriptive like error deleting droplet or something. this might have just been your debugging so possibly drop is better?

Reply:
Dropped. Yes, it was for debugging.